File: 卡码/代码随想录/1/06-二叉树/Python/01-二叉树遍历方式/LeetCode_0429.py
#429. N 叉树的层序遍历
from typing import *

class Node:
  def __init__(self, val=None, children=None):
    self.val = val
    self.children = children

class Solution:
  def levelOrder(self, root: 'Node') -> List[List[int]]:
    res:List[List[int]]=[]
    q:List[Node]=[]
    if root!=None:
      q.append(root)
    while len(q)>0:
      size:int=len(q)
      v:List[int]=[]
      for i in range(size):
        node:Node=q[0]
        q=q[1:]
        v.append(node.val)
        for child in node.children:
          q.append(child)
      res.append(v)
    return res

File: 卡码/代码随想录/1/06-二叉树/Go/01-二叉树遍历方式/LeetCode_0429.go
//429. N 叉树的层序遍历
type Node struct {
	Val int
	Children []*Node
}

func levelOrder(root *Node) [][]int {
	res:=make([][]int,0)
	q:=make([]*Node,0)
	var size int
	if root!=nil {
		q=append(q,root)
	}
	for len(q)>0 {
		size=len(q)
		v:=make([]int,0)
		for size>0 {
			size--
			node:=q[0]
			q=q[1:]
			v=append(v,node.Val)
			for _,tmp:=range node.Children {
				q=append(q,tmp)
			}
		}
		res=append(res,v)
	}
	return res
}
